### PR TITLE
No map collisions on FiltersTests

### PR DESCRIPTION
**Pull request by nik9000**
Adds randomUnique to generate unique things and uses it to make unique
keys.

The offending seed was 81AE616FEAD10F17.

### CODE REVIEW

**Comment by nik9000** [comment]
@cbuescher I remember you did a thing like this recently and I found another place where it is needed so I made a utility.

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java [hunk 1]
```diff
@@ -599,6 +600,22 @@ private static String groupName(ThreadGroup threadGroup) {
...
+    public static <T> Set<T> randomUnique(Supplier<T> supplier, int count) {
+        Set<T> things = new HashSet<>();
+        int maxTries = count * 10;
+        for (int t = 0; t < maxTries; t++) {
+            if (things.size() == count) {
+                return things;
+            }
+            things.add(supplier.get());
+        }
+        // Oh well, we didn't get enough unique things. It'll be ok.
```
Comment (cbuescher) [inline test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java]: I think this is okay as long as you clearly put that constraint into the comment. Maybe also call "count" something more approximate, like "itemsRequested" or sth. and clearly state that this many items will only be returned if a) the supplier can output enough random objects and b) we are lucky enough to get that many. I think for most tests this should be fine.

**Comment by cbuescher** [comment]
I like it, left one comment to be more clear about the limitations of the requested set size in comment and naming of the arguments, other than that LGTM

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java (hunk 1 above)
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java]: Will do! Do you remember any other spots we do this and I can switch those spots to here?

Comment (cbuescher) [inline test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java]: From the top of my head no, but I will keep it in my mind to change those when I see them.

**Pull request merged**